### PR TITLE
Minortopography/genpass

### PR DESCRIPTION
**Pull request by Omnielement**
add giant pine tree

### CODE REVIEW

File: Sources/Modules/Minortopography/GiantPinetree/Items/HarvestingClaw.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+		Item.axe = 1; // How much axe power the weapon has, note that the axe power displayed in-game is this value multiplied by 5
+	}
+	public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+	{
+		if (player.altFunctionUse == 2)
+		{
+			return false;
+		}
+		return base.Shoot(player, source, position, velocity, type, damage, knockback);
```
Comment (Setnour6) [inline Sources/Modules/Minortopography/GiantPinetree/Items/HarvestingClaw.cs:32]: When In tested this, the right click function does the same thing as the left click function but without auto use. How come there is no Item.shoot or other function for shoot?

File: Sources/Modules/Minortopography/GiantPinetree/Projectiles/PineSprite.png
Comment (Setnour6) [inline Sources/Modules/Minortopography/GiantPinetree/Projectiles/PineSprite.png:1]: The sprite should have a different sprite/image/texture. It does not look proper in appearance as just a green box.
image
You can still keep it as an extra sprite for highlighting the hostile NPCs. I will let you decide

File: Sources/Modules/Minortopography/GiantPinetree/Items/CreatPineTree.cs
Comment (Setnour6) [inline Sources/Modules/Minortopography/GiantPinetree/Items/CreatPineTree.cs:1]: This creates the giant pine cones and chest but not any trees

File: Sources/Modules/Minortopography/GiantPinetree/Items/SungloChandelier.cs
Comment (Setnour6) [inline Sources/Modules/Minortopography/GiantPinetree/Items/SungloChandelier.cs:1]: The animation for the chandelier does not work properly
image
Mining it also makes the game freeze/crash and closes the game.

**Review by Setnour6 — CHANGES_REQUESTED** [review]
So what needs to change is
- Iced Spear hurting town NPCs
- the pine sprite
- the sunglow chandelier's animation being broken
- mining the sunglow chandelier crashes the game

**Comment by Omnielement** [comment]
@Setnour6 What are you doing in IcedSpear(Projectile)? "Collide" is using to detect NPC, it's not a vanilla method.

**Comment by Setnour6** [comment]
> @Setnour6 What are you doing in IcedSpear(Projectile)? "Collide" is using to detect NPC, it's not a vanilla method.

There is an issue when I point the spear to the ground on a lower surface area and the spear is touching the ground. I wanted to fix that.
dotnet_IFbqBr1ifX

**Comment by Omnielement** [comment]
> > @Setnour6 What are you doing in IcedSpear(Projectile)? "Collide" is using to detect NPC, it's not a vanilla method.
> 
> There is an issue when I point the spear to the ground on a lower surface area and the spear is touching the ground. I wanted to fix that. dotnet_IFbqBr1ifX

Can it still stick to the target normally now?

**Comment by Setnour6** [comment]
> Can it still stick to the target normally now?

It looks like it can
dotnet_udZ0dNiZQ3

**Comment by Omnielement** [comment]
> > Can it still stick to the target normally now?
> 
> It looks like it can dotnet_udZ0dNiZQ3

I apologize, I was misunderstood your code.Everything was well, you can revert the "undo".

**Comment by Setnour6** [comment]
> I apologize, I was misunderstood your code.Everything was well, you can revert the "undo".

That is okay. I know you wanted the collision functions to work properly and I do too

**Comment by Setnour6** [comment]
I tested it. The last thing that needs to change is the sunglow chandelier animation not working, and that's it.
Also, is the pine sprite supposed to attack enemies or does it just hover over them to highlight them?

**Comment by Omnielement** [comment]
this branch still very important after this merge

**Pull request merged**